tools/cert-sidecar/README.md

```diff
@@ -0,0 +1,154 @@
+
+You can either use environment variables or configuration files to change their values.
+
+#### General
+
+General configurations
+
+| Key | Purpose | Default Value | Valid Values | Environment variable
+| --- | ------- | ------------- | ------------ | --------------------
+| app.sidecar.to      | Suffix to be used to identify logs | app | string | CERT_SC_APP_SIDECAR_TO
+| log.console.level   | Console logger level | info | info, debug, error, warn | CERT_SC_LOG_CONSOLE_LEVEL
+| log.console.verbose | Whether to enable logger verbosity or not | false | boolean | CERT_SC_LOG_CONSOLE_VERBOSE
+| log.file            | Enables logging on file (location: /var/log/cert-sidecar-logs-%DATE%.log) | false | boolean  | CERT_SC_LOG_FILE
+| log.file.level      | Log level to log on files | debug | string  | CERT_SC_LOG_LEVEL
+
+#### Certificates
+
+Certificates configurations
+
+| Key | Purpose | Default Value | Valid Values | Environment variable
+| --- | ------- | ------------- | ------------ | --------------------
+| certs.common.name   | The CN (common name). If the *HOSTNAME* environment variable is passed, this value will be ignored. | ${HOSTNAME:-generic-commonName} | string | CERT_SC_CERTS_COMMON_COMMON
+| certs.hostnames     | The Subject Alternative Name field | ["localhost"] | array of hostname/ip | CERT_SC_CERTS_HOSTNAMES
+| certs.expiration.checkend | The time window configured in seconds that defines whether the certificate is considered expired. If the time to expire is less than or equal to that time, it is considered expired. | 43200 | integer | CERT_SC_CERTS_EXPIRATION_CHECKEND
+| certs.crl | Enables the use of CRL (obtaining and creating the file)  | true | boolean | CERT_SC_CERTS_CRL
+| certs.files.basepath | Path to create files  | /certs | path | CERT_SC_CERTS_FILES_BASEPATH
+| certs.files.ca |  CA root certificate filename | ca.pem | filename  | CERT_SC_CERTS_FILES_CA
+| certs.files.cert | Public certificate filename | cert.pem | filename  | CERT_SC_CERTS_FILES_CERT
+| certs.files.crl | CRL filename  | crl.pem | filename | CERT_SC_CERTS_FILES_CRL
+| certs.files.key | Private keys filename | key.pem | filename | CERT_SC_CERTS_FILES_KEY
+
+#### Cron
+
+Cron configurations
+
+| Key | Purpose | Default Value | Valid Values | Environment variable
+| --- | ------- | ------------- | ------------ | --------------------
+| cron.crl | Enables the use of *cron* for updating CRL. | true | boolean | CERT_SC_CRON_CRL
+| cron.crl.time | Cron patterns for updating CRL. [Read up on cron patterns](http://crontab.org/) | 0 */2 * * * | string | CERT_SC_CRON_CRL_TIME
+| cron.expiration | Enables the use of *cron* for checking expiration for Public Certificate and Public CA certificate.  | true | boolean | CERT_SC_CRON_EXPIRATION
+| cron.expiration.time |  Cron patterns for checking expiration. [Read up on cron patterns](http://crontab.org/) | 0 1 * * * | string | CERT_SC_CRON_EXPIRATION_TIME
```

Comment:
See the note in the cron-node library:

`Read up on cron patterns here. Note the examples in the link have five fields, and 1 minute as the finest granularity, but this library has six fields, with 1 second as the finest granularity.`